test/src/nyc-test.js

```diff
@@ -83,6 +83,22 @@ describe('nyc', function () {
 
       nyc.exclude.length.should.eql(19)
     })
+
+    it("loads 'extensions' patterns from package.json#config.nyc", function () {
+      var nyc = new NYC({
+        cwd: path.resolve(__dirname, '../fixtures')
+      })
+
+      nyc.extensions.length.should.eql(2)
+    })
+
+    it("loads 'extensions' from package.json#nyc", function () {
```

Comment:
@bcoe I'm thinking this (and similar tests) should really be testing another fixture directory, rather than the top level `package.json`. Kinda weird to allow `.es6` for the nyc project itself.